src/test/java/jenkins/plugins/git/AbstractGitSCMSourceTest.java

```diff
@@ -302,11 +287,18 @@ public void retrieveTags_folderScopedCredentials() throws Exception {
         source.setTraits(new ArrayList<>());
         source.setCredentialsId(fCredentialsId);
 
-        SCMRevision rev = source.fetch("lightweight", listener, p);
-        assertThat(rev, notNullValue());
-        assertThat(rev.getHead().toString(), equalTo("SCMHead{'lightweight'}"));
-        Mockito.verify(gitClient, Mockito.times(0)).addDefaultCredentials(null);
-        Mockito.verify(gitClient, Mockito.atLeastOnce()).addDefaultCredentials(fCredentials);
+        Git git = Mockito.mock(Git.class, Mockito.CALLS_REAL_METHODS);
+        GitClient gitClient = Mockito.spy(git.getClient());
+        // Spy on GitClient methods
+        try (MockedStatic<Git> gitMock = Mockito.mockStatic(Git.class, Mockito.CALLS_REAL_METHODS)) {
+            gitMock.when(() -> Git.with(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(git);
+            doReturn(gitClient).when(git).getClient();
+            SCMRevision rev = source.fetch("lightweight", listener, p);
+            assertThat(rev, notNullValue());
+            assertThat(rev.getHead().toString(), equalTo("SCMHead{'lightweight'}"));
+            Mockito.verify(gitClient, Mockito.times(0)).addDefaultCredentials(null);
```

Comment:
Personally I would static import mockito and argument marchers as it leads to easier reading of the test imo, also given doReturn is static imported but the rest aren’t…